config/config.go

```diff
@@ -217,7 +216,7 @@ type ScrapeConfig struct {
 	// Indicator whether the scraped metrics should remain unmodified.
 	HonorLabels bool `yaml:"honor_labels,omitempty"`
 	// A set of query parameters with which the target is scraped.
-	Params url.Values `yaml:"params,omitempty"`
+	Params map[string]string `yaml:"params,omitempty"`
```

Comment:
Yes, but not sure whether people are generally aware of that .

Cannot really check right now, but are there no characters that cannot be percent-encoded that could serve as a separator?

The idea of numbering the label names really does not appeal to me. Making it truly non-breaking will be particularly ugly.

Let's think this through or explicitly be open to do something breaking in the future (which would be fine as this is a very advanced feature).